src/main/java/net/dv8tion/jda/client/entities/impl/ApplicationImpl.java

```diff
@@ -62,7 +63,7 @@ public ApplicationImpl(final JDA api, final JSONObject object)
         if (this.hasBot())
             return new RestAction.EmptyRestAction<>(this.bot);
 
-        return new RestAction<Application.Bot>(this.api, Route.Applications.CREATE_BOT.compile(this.id), null)
+        return new RestAction<Application.Bot>(this.api, Route.Applications.CREATE_BOT.compile(getId()), null)
```

Comment:
why don't you use the long for Routes? would need one string creation less per RestAction